src/v/cluster/archival_metadata_stm.cc

```diff
@@ -145,7 +145,13 @@ ss::future<std::error_code> archival_metadata_stm::do_add_segments(
     } else {
         fut = ss::with_timeout(deadline, std::move(fut));
     }
-    auto result = co_await std::move(fut);
+
+    result<raft::replicate_result> result{{}};
```

Comment:
is this constructed with a default initialized replicate_result because result<> doesn't allow default constrution?

Reply:
Yes. It will be updated later either way.